crates/sui-replay/src/data_fetcher.rs

```diff
@@ -773,4 +791,8 @@ impl DataFetcher for NodeStateDumpFetcher {
     ) -> Result<Vec<SuiEvent>, ReplayEngineError> {
         unimplemented!("get_epoch_change_events for state dump is not implemented")
     }
+
+    async fn get_chain_id(&self) -> Result<String, ReplayEngineError> {
+        unimplemented!("get_chain_id for state dump is not implemented")
+    }
```

Comment:
Curious what the complexities are here

Reply:
Well its really a serde thing.
Existing node dumps wont have the field, so they'll fail to deserialize if we impose the field.
Although now I'm thinking I can just use `#[serde(default...` brb

Reply:
This could be disruptive to core logic so I'll add support for that in subseq PR